simulationworkflowschema/equation_of_state.py

```diff
@@ -147,9 +150,100 @@ class EquationOfState(ParallelSimulation):
 
     results = SubSection(sub_section=EquationOfStateResults)
 
+    # def extract_indices_from_proxy_value(self, path: str):
+    #     """
+    #     Extracts run_index and system_index from a path string that contains '/run/<int>/system/<int>'.
+
+    #     Args:
+    #         path (str): The input path string.
+
+    #     Returns:
+    #         tuple: (run_index, system_index) as integers, or (None, None) if not found or parsing fails.
+    #     """
+    #     parts = path.split('/run/')
+    #     if len(parts) > 1:
+    #         try:
+    #             indices = parts[1].split('/system/')
+    #             run_index = int(indices[0])
+    #             system_index = int(indices[1])
+    #             return run_index, system_index
+    #         except (IndexError, ValueError):
+    #             return None, None
+    #     else:
+    #         return None, None
+
     def normalize(self, archive, logger):
         super().normalize(archive, logger)
 
+        print('in eos workflow normalizer')
+        # find and verify the input structure
+        if self.inputs:
+            input_structure = {}
+            for input_item in self.inputs:
+                # self.m_proxy_value = m_proxy_value
+                # self.m_proxy_section = m_proxy_section
+                # self.m_proxy_resolved = None
+                # self.m_proxy_type = m_proxy_type
+                # self.m_proxy_context = m_proxy_context
+                # print(input_item)
+                # logger.warning(f'input_item: {input_item}')
+                # print(f'input_item.m_proxy_value: {input_item.section}')
+                # logger.warning(f'input_item.m_proxy_value: {input_item.section}')
+                if isinstance(input_item.section, MProxy):
+                    input_section = input_item.section.m_proxy_resolve()
+                    input_proxy_value = input_item.section.m_proxy_value
+                    # input_system = input_item.section.m_proxy_resolved
+                    # run_index, system_index = self.extract_indices_from_proxy_value(
+                    #     input_item.section.m_proxy_value
+                    # )
+                else:
+                    input_section = input_item.section
+                    input_proxy_value = ''
+                #     input_system = input_item.section
+                system_index = input_section.m_parent_index
+                run_section = input_section.m_parent
+                run_index = run_section.m_parent_index
+                # logger.warning(
+                #     f'input_system: {input_system}, system_index: {system_index}'
+                # )
+                # logger.warning(f'run_section: {run_section}, run_index: {run_index}')
+                # run_index, system_index = ...?
+                if isinstance(input_section, System):
+                    if input_structure:
```

Comment:
do we really need to loop over all inputs or simply break once we get the one with the system info?

Reply:
It's only because I thought it would be nice to return some indication that we were taking the first structure in case there were several structures included as input...you think this is overkill?

Reply:
ok maybe you can filter all system inputs and put the log if there are multiple

Reply:
The user does have control if they are uploading a workflow yaml, or are we thinking of different things here?

- Yes 100% agree about the docs in either case. It seems that we (also Julia in the neb parser) are moving toward more automated generation of the graph structure within these normalization functions with the idea that these may be instantiated from a parser OR from a yaml. In this case it must be very clear what these normalizers are doing I think

Reply:
> different

i meant that we picking the first system and not the others that was provided.